src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -285,11 +294,10 @@ trait Unapplies extends ast.TreeDSL {
         if (currentRun.isScala3) {
           val inheritedMods = constrMods(cdef)
           val mods3 = Modifiers(SYNTHETIC | (inheritedMods.flags & AccessFlags), inheritedMods.privateWithin)
-          if (currentRun.isScala3Cross)
-            mods3
+          if (currentRun.scala3Features.caseApplyCopyAccess) mods3
           else {
             if (mods3 != synth)
-              runReporting.warning(cdef.pos, "access modifiers for `copy` method are copied from the case class constructor", Scala3Migration, cdef.symbol)
+              runReporting.warning(cdef.namePos, "access modifiers for `copy` method will be copied from the case class constructor under Scala 3", Scala3Migration, cdef.symbol)
```

Comment:
this was a tweak to tighten the span of the diagnostic. Eventually someone using metals would notice.

The apply method is still just the tree focus because I was too lazy to downcast. Or I was too downcast.